src/lib/containers/download_list/CreatePackage.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from 'react'
+import {
+  getFileHandleByIdURL,
+  getDownloadOrder,
+  getBulkFiles,
+} from '../../utils/SynapseClient'
+import {
+  BulkFileDownloadRequest,
+  ZipFileFormat,
+} from '../../utils/jsonResponses/BulkFileDownloadRequest'
+import { BulkFileDownloadResponse } from '../../utils/jsonResponses/BulkFileDownloadResponse'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { library } from '@fortawesome/fontawesome-svg-core'
+import {
+  faCheck,
+  faDownload,
+  faFolder,
+} from '@fortawesome/free-solid-svg-icons'
+import * as ReactBoostrap from 'react-bootstrap'
+import { FileDownloadStatus } from '../../utils/jsonResponses/FileDownloadSummary'
+
+library.add(faCheck)
+library.add(faDownload)
+library.add(faFolder)
+
+export type CreatePackageProps = {
+  token?: string
+  children?: JSX.Element
+  updateDownloadList: Function
+}
+
+type Alert = {
+  message: string
+  variant: ReactBoostrap.AlertProps['variant']
+  className: string | undefined
+}
+
+export const templateDownload =
+  'files were downloaded and removed from the list.'
+export const templateErrorFileName =
+  'Please provide a package file name and try again.'
+
+export const CreatePackage = (props: CreatePackageProps) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const [fileName, setZipFileName] = useState('')
+  const [alert, setAlert] = useState<Alert>({
+    message: '',
+    className: undefined,
+    variant: undefined,
+  })
+  const [bulkFileDownloadResponse, setBulkFileDownloadResponse] = useState<
+    BulkFileDownloadResponse | undefined
+  >(undefined)
+  const { token, children, updateDownloadList } = props
+
+  const createPackageHandler = async (event: React.SyntheticEvent) => {
+    event && event.preventDefault()
```

Comment:
isn't event always defined ?

Reply:
In the testing environment its not, but perhaps there's a library that would help with that. For now I changed the tests to pass in a stub for preventDefault when triggering this method, which I think is preferable because then we don't have odd statements like the above, thanks for pointing this out.